src/transaction/components/TransactionProvider.tsx

```diff
@@ -68,7 +74,24 @@ export function TransactionProvider({
 
   const { switchChainAsync } = useSwitchChain();
 
-  // Hooks that depend from Core Hooks
+  // Validate `calls` and `contracts` props
+  if (!contracts && !calls) {
+    setErrorMessage('One of contracts or calls must be provided.');
+    return;
+  }
+  if (calls && contracts) {
+    setErrorMessage('Only one of contracts or calls can be provided.');
+    return;
+  }
+  // Sets transactions
+  useEffect(() => {
+    setTransactions(calls || contracts || []);
+  }, [calls, contracts]);
+
+  /*
+    useWriteContracts or useWriteContract
+    Used for contract calls with an ABI and functions.
```

Comment:
Multi-line comments are only for functions. For code inside functions we use single line comments. That's a common best practice in Typescript.

so do

```
//  useWriteContracts or useWriteContract
//  Used for contract calls with an ABI and functions.
```